static/panes/tool.js

```diff
@@ -215,16 +215,16 @@ Tool.prototype.getInputArgs = function () {
     }
 };
 
-Tool.prototype.onToolInputChange = function (compilerId, editorId, toolId, input) {
-    if (this.compilerId === compilerId && this.editorId === editorId && this.toolId === toolId) {
+Tool.prototype.onToolInputChange = function (compilerId, toolId, input) {
+    if (this.compilerId === compilerId && this.toolId === toolId) {
         this.monacoStdinField = input;
         this.onOptionsChange();
         this.eventHub.emit('toolSettingsChange', this.compilerId);
     }
 };
 
-Tool.prototype.onToolInputViewClosed = function (compilerId, editorId, toolId, input) {
-    if (this.compilerId === compilerId && this.editorId === editorId && this.toolId === toolId) {
+Tool.prototype.onToolInputViewClosed = function (compilerId, toolId, input) {
+    if (this.compilerId === compilerId && this.toolId === toolId) {
```

Comment:
This, and the onToolInputChange is the only thing that had me thinking. But I've convinced myself it's OK :)

Reply:
considering compilerId's (pane id's) are unique, it should be fine